controllers/articles.controllers.js

```diff
@@ -76,3 +76,28 @@ exports.getArticles = (request, response, next) => {
         })
         .catch(next);
 };
+
+exports.getCommentsByArticleId = (request, response, next) => {
+    const { article_id } = request.params;
+
+    if (isNaN(article_id) || !Number.isInteger(Number(article_id))) {
+        return response.status(400).send({ message: "Invalid ID" });
+    }
+
+    const id = Number(article_id);
+
+    selectArticleById(id)
+        .then(() => {
+        return selectCommentsByArticleId(id);
+        })
+        .then(comments => {
+            response.status(200).send({ comments });
+        })
+        .catch((err) => {
+            if (err.status) {
+                response.status(err.status).send({ message: err.message });
```

Comment:
Your middleware in your app handles this. Having it here means your code isn't dry. How can you pass it to your error handling middleware?